pkgs/servers/http/apache-modules/mod_tile/default.nix

```diff
@@ -1,77 +1,68 @@
-{ lib
+{ fetchFromGitHub
+, lib
 , stdenv
-, fetchFromGitHub
-, fetchpatch
-, autoreconfHook
 , apacheHttpd
 , apr
+, aprutil
+, boost
 , cairo
+, cmake
+, curl
+, glib
+, gtk2
+, harfbuzz
+, icu
 , iniparser
+, libmemcached
 , mapnik
-, boost
-, icu
-, harfbuzz
-, libjpeg
-, libtiff
-, libwebp
-, proj
-, sqlite
 }:
 
 stdenv.mkDerivation rec {
   pname = "mod_tile";
-  version = "unstable-2017-01-08";
+  version = "0.6.1+unstable=2023-03-09";
 
   src = fetchFromGitHub {
     owner = "openstreetmap";
     repo = "mod_tile";
-    rev = "e25bfdba1c1f2103c69529f1a30b22a14ce311f1";
-    sha256 = "12c96avka1dfb9wxqmjd57j30w9h8yx4y4w34kyq6xnf6lwnkcxp";
+    rev = "f521540df1003bb000d7367a59ad612161eab0f0";
+    sha256 = "sha256-jIqeplAQt4W97PNKm6ZDGPDUc/PEiLM5yEdPeI+H03A=";
   };
 
-  patches = [
-    # Pull upstream fix for -fno-common toolchains:
-    #  https://github.com/openstreetmap/mod_tile/pull/202
-    (fetchpatch {
-      name = "fno-common";
-      url = "https://github.com/openstreetmap/mod_tile/commit/a22065b8ae3c018820a5ca9bf8e2b2bb0a0bfeb4.patch";
-      sha256 = "1ywfa14xn9aa96vx1adn1ndi29qpflca06x986bx9c5pqk761yz3";
-    })
+  nativeBuildInputs = [
+    cmake
   ];
 
-  # test is broken and I couldn't figure out a better way to disable it.
-  postPatch = ''
-    echo "int main(){return 0;}" > src/gen_tile_test.cpp
-  '';
-
-  nativeBuildInputs = [ autoreconfHook ];
   buildInputs = [
     apacheHttpd
     apr
+    aprutil
+    boost
     cairo
+    curl
+    glib
+    harfbuzz
+    icu
     iniparser
+    libmemcached
     mapnik
-    boost
-    icu
-    harfbuzz
-    libjpeg
-    libtiff
-    libwebp
-    proj
-    sqlite
   ];
 
-  configureFlags = [
-    "--with-apxs=${apacheHttpd.dev}/bin/apxs"
-  ];
-
-  installPhase = ''
-    mkdir -p $out/modules
-    make install-mod_tile DESTDIR=$out
-    mv $out${apacheHttpd}/* $out
-    rm -rf $out/nix
+  # the install script wants to install mod_tile.so into apache's modules dir
+  postPatch = ''
+    sed -i "s|\''${HTTPD_MODULES_DIR}|$out/modules|" CMakeLists.txt
   '';
 
+  # Although glib is detected in the configure phase, due to nix derivation
+  # output splitting, there are header files that are in glib's `out` output and
+  # not in the `dev` output.
+  NIX_CFLAGS_COMPILE = "-I${glib.out}/lib/glib-2.0/include";
```

Comment:
`glib` detection should still Just Work in `cmake`. It fails because `pkg-config` tool that `cmake` usually uses was not available. The following seems to fix the build for me:

```diff
--- a/pkgs/servers/http/apache-modules/mod_tile/default.nix
+++ b/pkgs/servers/http/apache-modules/mod_tile/default.nix
@@ -15,6 +15,7 @@
 , iniparser
 , libmemcached
 , mapnik
+, pkg-config
 }:

 stdenv.mkDerivation rec {
@@ -30,6 +31,7 @@ stdenv.mkDerivation rec {

   nativeBuildInputs = [
     cmake
+    pkg-config
   ];

   buildInputs = [
@@ -55,7 +57,7 @@ stdenv.mkDerivation rec {
   # Although glib is detected in the configure phase, due to nix derivation
   # output splitting, there are header files that are in glib's `out` output and
   # not in the `dev` output.
-  NIX_CFLAGS_COMPILE = "-I${glib.out}/lib/glib-2.0/include";
+  #NIX_CFLAGS_COMPILE = "-I${glib.out}/lib/glib-2.0/include";
   enableParallelBuilding = true;

   # We need to either disable the `render_speedtest` and `download_tile` tests
```

Reply:
You're right, that does the job! Thank you.